### ISSUE TITLE
[FVT]updatestatus is always failed on Ubuntu16.04 and xdsh plugin bug

### ISSUE THREAD

**Issue by tingtli** [xcat2/xcat-core#1160]
xCAT 2.12 Ubuntu16.4 ppc64le
```
root@c910f03c17k27:/# updatenode c910f03c05k51 -V
Running command on c910f03c17k27: /bin/hostname 2>&1

Running command on c910f03c17k27: ip -4 --oneline addr show |awk -F ' ' '{print $4}'|awk -F '/' '{print $1}' 2>&1

Running command on c910f03c17k27: /bin/hostname 2>&1

Running command on c910f03c17k27: ip -4 --oneline addr show |awk -F ' ' '{print $4}'|awk -F '/' '{print $1}' 2>&1

Running command on c910f03c17k27: chmod -R a+r /install/postscripts 2>&1

  c910f03c17k27: Internal call command: xdcp c910f03c05k51 --nodestatus -F /install/custom/netboot/ubuntu/compute.synclist
Running internal xCAT command: xdcp ...

Running command on c910f03c17k27: /bin/hostname 2>&1

Running command on c910f03c17k27: ip -4 --oneline addr show |awk -F ' ' '{print $4}'|awk -F '/' '{print $1}' 2>&1

c910f03c05k51: xcatd: xdsh plugin bug, pid 11015, process description: 'xcatd SSL: updatenode to c910f03c05k51 for root@localhost: updatenode instance: xdsh instance' with error 'Can't use an undefined value as a symbol reference at /opt/xcat/lib/perl/xCAT/DSHCLI.pm line 918.
c910f03c05k51: ' while trying to fulfill request for the following nodes: c910f03c05k51
File synchronization has completed for nodes.
Performing software maintenance operations. This could take a while, if there are packages to install.

  c910f03c17k27: Internal call command: xdsh c910f03c05k51 --nodestatus -s -v -e /install/postscripts/xcatdsklspost 2 -m 10.3.17.27 'ospkgs,otherpkgs,syscloneimgupdate' --tftp /tftpboot --installdir /install --nfsv4 no -c -V
Running command on c910f03c17k27: /bin/hostname 2>&1
Running command on c910f03c17k27: ip -4 --oneline addr show |awk -F ' ' '{print $4}'|awk -F '/' '{print $1}' 2>&1
Running command on c910f03c17k27: hostname 2>&1
Running command on c910f03c17k27: /opt/xcat/bin/pping c910f03c05k51 2>&1
Error: Error from pping
c910f03c05k51: running /tmp/filePe6GGz.dsh 2 -m 10.3.17.27 ospkgs,otherpkgs,syscloneimgupdate --tftp /tftpboot --installdir /install --nfsv4 no -c -V
c910f03c05k51: xcatdsklspost : trying to download postscripts from http://10.3.17.27/install/postscripts/
c910f03c05k51: /tmp/filePe6GGz.dsh : download_postscripts return successfully
c910f03c05k51: xcatdsklspost: downloaded postscripts successfully
c910f03c05k51: /tmp/filePe6GGz.dsh :  trying to download http://10.3.17.27/tftpboot/mypostscripts/mypostscript.c910f03c05k51
c910f03c05k51: /tmp/filePe6GGz.dsh download_mypostscript return successfully
c910f03c05k51: running //xcatpost/mypostscript
c910f03c05k51: Thu May 19 23:13:39 EDT 2016 Running postscript: ospkgs
c910f03c05k51: NFSSERVER=10.3.17.27
c910f03c05k51: OSPKGDIR=/install/ubuntu16.04/ppc64el
c910f03c05k51: OSPKGS = bash,nfs-common,openssl,isc-dhcp-client,libc-bin,openssh-server,openssh-client,wget,vim,ntp,ntpdate,rsync,busybox-static,gawk,dnsutils
c910f03c05k51: pkgs= bash nfs-common openssl isc-dhcp-client libc-bin openssh-server openssh-client wget vim ntp ntpdate rsync busybox-static gawk dnsutils
c910f03c05k51: groups=
c910f03c05k51: remove groups=
c910f03c05k51: remove pkgs=
c910f03c05k51: deb http://10.3.17.27/install/ubuntu16.04/ppc64el xenial main restricted
c910f03c05k51: deb http://10.3.17.27/install/ubuntu16.04/ppc64el xenial-updates main restricted
c910f03c05k51: Ign:1 http://10.3.17.27/install/ubuntu16.04/ppc64el xenial InRelease
c910f03c05k51: Ign:2 http://10.3.17.27/install/ubuntu16.04/ppc64el xenial-updates InRelease
c910f03c05k51: Hit:3 http://10.3.17.27/install/ubuntu16.04/ppc64el xenial Release
c910f03c05k51: Ign:4 http://10.3.17.27/install/ubuntu16.04/ppc64el xenial-updates Release
c910f03c05k51: Ign:5 http://10.3.17.27/install/ubuntu16.04/ppc64el xenial-updates/main ppc64el Packages
c910f03c05k51: Ign:6 http://10.3.17.27/install/ubuntu16.04/ppc64el xenial-updates/main all Packages
c910f03c05k51: Ign:7 http://10.3.17.27/install/ubuntu16.04/ppc64el xenial-updates/main Translation-en
c910f03c05k51: Ign:8 http://10.3.17.27/install/ubuntu16.04/ppc64el xenial-updates/restricted ppc64el Packages
c910f03c05k51: Ign:9 http://10.3.17.27/install/ubuntu16.04/ppc64el xenial-updates/restricted all Packages
c910f03c05k51: Ign:10 http://10.3.17.27/install/ubuntu16.04/ppc64el xenial-updates/restricted Translation-en
c910f03c05k51: Ign:5 http://10.3.17.27/install/ubuntu16.04/ppc64el xenial-updates/main ppc64el Packages
c910f03c05k51: Ign:6 http://10.3.17.27/install/ubuntu16.04/ppc64el xenial-updates/main all Packages
c910f03c05k51: Ign:7 http://10.3.17.27/install/ubuntu16.04/ppc64el xenial-updates/main Translation-en
c910f03c05k51: Ign:8 http://10.3.17.27/install/ubuntu16.04/ppc64el xenial-updates/restricted ppc64el Packages
c910f03c05k51: Ign:9 http://10.3.17.27/install/ubuntu16.04/ppc64el xenial-updates/restricted all Packages
c910f03c05k51: Ign:10 http://10.3.17.27/install/ubuntu16.04/ppc64el xenial-updates/restricted Translation-en
c910f03c05k51: Ign:5 http://10.3.17.27/install/ubuntu16.04/ppc64el xenial-updates/main ppc64el Packages
c910f03c05k51: Ign:6 http://10.3.17.27/install/ubuntu16.04/ppc64el xenial-updates/main all Packages
c910f03c05k51: Ign:7 http://10.3.17.27/install/ubuntu16.04/ppc64el xenial-updates/main Translation-en
c910f03c05k51: Ign:8 http://10.3.17.27/install/ubuntu16.04/ppc64el xenial-updates/restricted ppc64el Packages
c910f03c05k51: Ign:9 http://10.3.17.27/install/ubuntu16.04/ppc64el xenial-updates/restricted all Packages
c910f03c05k51: Ign:10 http://10.3.17.27/install/ubuntu16.04/ppc64el xenial-updates/restricted Translation-en
c910f03c05k51: Ign:5 http://10.3.17.27/install/ubuntu16.04/ppc64el xenial-updates/main ppc64el Packages
c910f03c05k51: Ign:6 http://10.3.17.27/install/ubuntu16.04/ppc64el xenial-updates/main all Packages
c910f03c05k51: Ign:7 http://10.3.17.27/install/ubuntu16.04/ppc64el xenial-updates/main Translation-en
c910f03c05k51: Ign:8 http://10.3.17.27/install/ubuntu16.04/ppc64el xenial-updates/restricted ppc64el Packages
c910f03c05k51: Ign:9 http://10.3.17.27/install/ubuntu16.04/ppc64el xenial-updates/restricted all Packages
c910f03c05k51: Ign:10 http://10.3.17.27/install/ubuntu16.04/ppc64el xenial-updates/restricted Translation-en
c910f03c05k51: Ign:5 http://10.3.17.27/install/ubuntu16.04/ppc64el xenial-updates/main ppc64el Packages
c910f03c05k51: Ign:6 http://10.3.17.27/install/ubuntu16.04/ppc64el xenial-updates/main all Packages
c910f03c05k51: Ign:7 http://10.3.17.27/install/ubuntu16.04/ppc64el xenial-updates/main Translation-en
c910f03c05k51: Ign:8 http://10.3.17.27/install/ubuntu16.04/ppc64el xenial-updates/restricted ppc64el Packages
c910f03c05k51: Ign:9 http://10.3.17.27/install/ubuntu16.04/ppc64el xenial-updates/restricted all Packages
c910f03c05k51: Ign:10 http://10.3.17.27/install/ubuntu16.04/ppc64el xenial-updates/restricted Translation-en
c910f03c05k51: Err:5 http://10.3.17.27/install/ubuntu16.04/ppc64el xenial-updates/main ppc64el Packages
c910f03c05k51:   404  Not Found
c910f03c05k51: Ign:6 http://10.3.17.27/install/ubuntu16.04/ppc64el xenial-updates/main all Packages
c910f03c05k51: Ign:7 http://10.3.17.27/install/ubuntu16.04/ppc64el xenial-updates/main Translation-en
c910f03c05k51: Err:8 http://10.3.17.27/install/ubuntu16.04/ppc64el xenial-updates/restricted ppc64el Packages
c910f03c05k51:   404  Not Found
c910f03c05k51: Ign:9 http://10.3.17.27/install/ubuntu16.04/ppc64el xenial-updates/restricted all Packages
c910f03c05k51: Ign:10 http://10.3.17.27/install/ubuntu16.04/ppc64el xenial-updates/restricted Translation-en
c910f03c05k51: Hit:12 http://ports.ubuntu.com/ubuntu-ports xenial-security InRelease
c910f03c05k51: Reading package lists...
c910f03c05k51: W: The repository 'http://10.3.17.27/install/ubuntu16.04/ppc64el xenial-updates Release' does not have a Release file.
c910f03c05k51: E: Failed to fetch http://10.3.17.27/install/ubuntu16.04/ppc64el/dists/xenial-updates/main/binary-ppc64el/Packages  404  Not Found
c910f03c05k51: E: Failed to fetch http://10.3.17.27/install/ubuntu16.04/ppc64el/dists/xenial-updates/restricted/binary-ppc64el/Packages  404  Not Found
c910f03c05k51: E: Some index files failed to download. They have been ignored, or old ones used instead.
c910f03c05k51: === DEBIAN_FRONTEND=noninteractive apt-get -y --allow-unauthenticated --force-yes -o Dpkg::Options::='--force-confold' -o Dpkg::Options::='--force-confdef' upgrade
c910f03c05k51: Reading package lists...
c910f03c05k51: Building dependency tree...
c910f03c05k51: Reading state information...
c910f03c05k51: Calculating upgrade...
c910f03c05k51: 0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
c910f03c05k51: ===  apt-get -q -y --force-yes install --no-install-recommends  bash nfs-common openssl isc-dhcp-client libc-bin openssh-server openssh-client wget vim ntp ntpdate rsync busybox-static gawk dnsutils
c910f03c05k51: Reading package lists...
c910f03c05k51: Building dependency tree...
c910f03c05k51: Reading state information...
c910f03c05k51: bash is already the newest version (4.3-14ubuntu1).
c910f03c05k51: dnsutils is already the newest version (1:9.10.3.dfsg.P4-8).
c910f03c05k51: busybox-static is already the newest version (1:1.22.0-15ubuntu1).
c910f03c05k51: gawk is already the newest version (1:4.1.3+dfsg-0.1).
c910f03c05k51: nfs-common is already the newest version (1:1.2.8-9ubuntu12).
c910f03c05k51: ntp is already the newest version (1:4.2.8p4+dfsg-3ubuntu5).
c910f03c05k51: ntpdate is already the newest version (1:4.2.8p4+dfsg-3ubuntu5).
c910f03c05k51: openssh-client is already the newest version (1:7.2p2-4).
c910f03c05k51: openssh-server is already the newest version (1:7.2p2-4).
c910f03c05k51: rsync is already the newest version (3.1.1-3ubuntu1).
c910f03c05k51: vim is already the newest version (2:7.4.1689-3ubuntu1).
c910f03c05k51: wget is already the newest version (1.17.1-1ubuntu1).
c910f03c05k51: openssl is already the newest version (1.0.2g-1ubuntu4.1).
c910f03c05k51: isc-dhcp-client is already the newest version (4.3.3-5ubuntu12).
c910f03c05k51: libc-bin is already the newest version (2.23-0ubuntu3).
c910f03c05k51: 0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
c910f03c05k51: Postscript: ospkgs exited with code 0
c910f03c05k51: Thu May 19 23:13:40 EDT 2016 Running postscript: otherpkgs
c910f03c05k51: ./otherpkgs: no extra rpms to install
c910f03c05k51: Postscript: otherpkgs exited with code 0
c910f03c05k51: Thu May 19 23:13:40 EDT 2016 Running postscript: syscloneimgupdate
c910f03c05k51: Postscript: syscloneimgupdate exited with code 0
c910f03c05k51: //xcatpost/mypostscript return with 0
c910f03c05k51: Running of Software Maintenance has completed.
  c910f03c17k27: Internal call command: xdsh c910f03c05k51 --nodestatus -s -v -e /install/postscripts/xcatdsklspost 1 -m 10.3.17.27 '' --tftp /tftpboot --installdir /install --nfsv4 no -c -V
Running command on c910f03c17k27: /bin/hostname 2>&1
Running command on c910f03c17k27: ip -4 --oneline addr show |awk -F ' ' '{print $4}'|awk -F '/' '{print $1}' 2>&1
Running command on c910f03c17k27: hostname 2>&1
Running command on c910f03c17k27: /opt/xcat/bin/pping c910f03c05k51 2>&1
Error: Error from pping
c910f03c05k51: running /tmp/fileLXbYv9.dsh 1 -m 10.3.17.27  --tftp /tftpboot --installdir /install --nfsv4 no -c -V
c910f03c05k51: xcatdsklspost : trying to download postscripts from http://10.3.17.27/install/postscripts/
c910f03c05k51: /tmp/fileLXbYv9.dsh : download_postscripts return successfully
c910f03c05k51: xcatdsklspost: downloaded postscripts successfully
c910f03c05k51: /tmp/fileLXbYv9.dsh :  trying to download http://10.3.17.27/tftpboot/mypostscripts/mypostscript.c910f03c05k51
c910f03c05k51: /tmp/fileLXbYv9.dsh download_mypostscript return successfully
c910f03c05k51: running //xcatpost/mypostscript
c910f03c05k51: Thu May 19 23:13:42 EDT 2016 Running postscript: remoteshell
c910f03c05k51:
c910f03c05k51: root      1701  0.0  0.3  20992 14272 ?        Ss   23:10   0:00 sshd: root@pts/0
c910f03c05k51: root      2697  1.7  0.3  20992 14208 ?        Ss   23:13   0:00 sshd: root@notty
c910f03c05k51: root      2873  6.0  0.2  15360 10688 ?        Ss   23:13   0:00 /usr/sbin/sshd -D
c910f03c05k51: Postscript: remoteshell exited with code 0
c910f03c05k51: Thu May 19 23:13:44 EDT 2016 Running postscript: syncfiles
c910f03c05k51: Postscript: syncfiles exited with code 0
c910f03c05k51: Thu May 19 23:13:44 EDT 2016 Running postscript: otherpkgs
c910f03c05k51: ./otherpkgs: no extra rpms to install
c910f03c05k51: Postscript: otherpkgs exited with code 0
c910f03c05k51: //xcatpost/mypostscript return with 0
c910f03c05k51: Running of postscripts has completed.
root@c910f03c17k27:/# lsdef -l c910f03c05k51
Object name: c910f03c05k51
    arch=ppc64el
    cons=kvm
    currchain=boot
    currstate=boot
    groups=all.wjx
    initrd=xcat/osimage/ubuntu16.04-ppc64el-install-compute/initrd.img
    installnic=mac
    ip=10.3.5.51
    kcmdline=nofb utf8 auto url=http://!myipfn!/install/autoinst/c910f03c05k51 xcatd=!myipfn! mirror/http/hostname=!myipfn! netcfg/choose_interface=42:53:0a:03:05:33 console=tty0 console=hvc0,115200 locale=en_US priority=critical hostname=c910f03c05k51 live-installer/net-image=http://!myipfn!/install/ubuntu16.04/ppc64el/install/filesystem.squashfs
    kernel=xcat/osimage/ubuntu16.04-ppc64el-install-compute/vmlinuz
    mac=42:53:0a:03:05:33|42:d4:0a:03:05:33!*NOIP*
    mgt=kvm
    netboot=grub2
    os=ubuntu16.04
    postbootscripts=otherpkgs
    postscripts=remoteshell,syncfiles
    primarynic=eth0
    profile=compute
    provmethod=ubuntu16.04-ppc64el-install-compute
    serialport=0
    serialspeed=115200
    status=booted
    statustime=05-19-2016 23:07:44
    updatestatus=failed
    updatestatustime=05-19-2016 23:13:44
    vmcpus=1
    vmhost=c910f03c05
    vmmemory=4096
    vmnicnicmodel=virtio
    vmnics=br10,br50
    vmstorage=dir:///var/lib/libvirt/images/
```

**Comment by daniceexi** [comment]
this issue has been fix by the pull request #1165

**Comment by neo954** [comment]
Verified with xCAT 2.12-snap201605200311
